DataFormats/SiStripCluster/src/SiStripCluster.cc

```diff
@@ -2,28 +2,30 @@
 #include "DataFormats/SiStripCluster/interface/SiStripCluster.h"
 
 SiStripCluster::SiStripCluster(const SiStripDigiRange& range) : firstStrip_(range.first->strip()), error_x(-99999.9) {
-  amplitudes_.reserve(range.second - range.first);
+  std::vector<uint8_t> v;
```

Comment:
I may have misunderstood the code, but here a naive question: why not filling the *amplitudes_* vector directly without the temporary *v* vector?

Reply:
because makes no difference? (and avoid a dangerous interface...)

Reply:
I think this changes anticipates changing the underlying storage to a `std:array`, but it's not necessary indeed. Should I revert this for now? I don't know if it would be optimised out by the compiler.

Reply:
this constructor is absolutely irrelevant for any production wf
https://cmssdt.cern.ch/dxr/CMSSW/search?q=%2Bcallers%3A%22SiStripCluster%3A%3ASiStripCluster%28const+SiStripCluster%3A%3ASiStripDigiRange+%26%29%22